File: app/RegisterActivity.java
package com.yourpackage

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class RegisterActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val name = findViewById<EditText>(R.id.etName)
                val email = findViewById<EditText>(R.id.etEmail)
                val password = findViewById<EditText>(R.id.etPassword)
                val registerBtn = findViewById<Button>(R.id.btnRegister)

                registerBtn.setOnClickListener {
            val userName = name.text.toString()
            val userEmail = email.text.toString()

            Log.d("REGISTER", "User: $userName, Email: $userEmail")
            Toast.makeText(this, "Register berhasil untuk $userName", Toast.LENGTH_SHORT).show()
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }
}

File: app/src/main/java/com/example/uts_pm/RegisterActivity.kt
package com.example.uts_pm

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class RegisterActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val etName = findViewById<EditText>(R.id.etName)
        val etEmail = findViewById<EditText>(R.id.etEmail)
        val etPassword = findViewById<EditText>(R.id.etPassword)
        val etConfirmPassword = findViewById<EditText>(R.id.etConfirmPassword)
        val btnRegister = findViewById<Button>(R.id.btnRegister)
        val btnBackToLogin = findViewById<Button>(R.id.btnBackToLogin)

        btnRegister.setOnClickListener {
            Log.d("RegisterActivity", "Tombol Register ditekan")
            Toast.makeText(this, "Register ditekan", Toast.LENGTH_SHORT).show()

            val name = etName.text.toString()
            val email = etEmail.text.toString()
            val password = etPassword.text.toString()
            val confirm = etConfirmPassword.text.toString()

            if (name.isNotEmpty() && email.isNotEmpty() && password == confirm) {
                val sharedPref = getSharedPreferences("UserData", MODE_PRIVATE)
                with(sharedPref.edit()) {
                    putString("email", email)
                    putString("password", password)
                    apply()
                }
                Toast.makeText(this, "Berhasil daftar, silakan login", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this, LoginActivity::class.java))
                finish()
            }

        }

        btnBackToLogin.setOnClickListener {
            Log.d("RegisterActivity", "Kembali ke Login")
            finish()
        }
    }
}